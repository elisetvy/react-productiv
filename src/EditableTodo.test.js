import React from "react";
import { render } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

describe("Editable todo component", function (){
  const initialTodo= {
      id: 1,
      title: "Code!",
      description: "Write some code",
      priority: 2,
    }

  test("Should render without crashing", function () {
    render(<EditableTodo
      todo={initialTodo}
      update={() => console.log("test update")}
      remove={() => console.log("test remove")} />);
  });

  test("contains expected title", function () {
    const result = render(<TopTodo todos={initialTodos} />);
    expect(result.queryByText("Cook something healthy")).toBeInTheDocument();
  });

  test("passes snapshot", function (){
    const { container } = render(<TopTodo todos={initialTodos} />);
    expect(container).toMatchSnapshot();
  });
});