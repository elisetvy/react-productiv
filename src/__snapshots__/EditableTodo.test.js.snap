// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top todo component passes snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="2"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</div>
`;
